Compiler/src/implementation/Scanner.cpp

```diff
@@ -227,6 +247,87 @@ Token Scanner::consumeZeroToken()
 	return Token(INTEGER, "0");
 }
 
+Token Scanner::consumeGenericWordToken()
+{
+	string Accumulator = string(1, FileContents[FilePosition]);
+
+	//Set initial endState To False
+	bool ValidEndState = false;
+	while (!ValidEndState && FilePosition < FileContents.size())
+	{
+		if (FilePosition + 1 == FileContents.size())
+		{
+			//Last character in file. Update accumulator, and set End State To Valid.
+			FilePosition++;
+			ValidEndState = true;
+			continue;
+		}
+
+		char NextChar = FileContents[FilePosition + 1];
+
+
+		//If not the last character in file, check the next char.
+		//Also keep making sure that Accumulator doesn't exceed more than 256 characters.
+
+
+		if ((isalpha(NextChar) || NextChar == '_' || isdigit(NextChar)) && Accumulator.length() < 257) {
+			//If Next character a digit or an underscore and is of valid length, add to accumulator and keep going
+			Accumulator += NextChar;
+			FilePosition++;
+			continue;
+		}
+		else if (SelfDelimiters.find(NextChar) != string::npos || isspace(NextChar))
+		{
+			//If next character is a self delimiter, or a space, we're in a stop state.
+			ValidEndState = true;
+			FilePosition++;
+			continue;
+		}
+		else
+		{
+			//If any other character is recognized, blow up.
+			string ErrorMessage = "ERROR: Unexpected character while scanning Identifier at pos - " + to_string(FilePosition + 1) + " char= " + NextChar;
+			throw runtime_error(ErrorMessage);
+		}
+	}
+
+	//If the identifier is longer than 256 valid characters, tell them they can't
+	if (Accumulator.length() > 256) {
+		string ErrorMessage = "ERROR: Length of Identifier is toooo long at pos - " + to_string(FilePosition);
+		throw runtime_error(ErrorMessage);
+	}
+	else {
+			if (Accumulator == "function" || Accumulator == "main" || Accumulator == "print") {
+				FilePosition = FilePosition++;
+				return Token(PRIMITIVE_KEYWORD, Accumulator);
+			}
+			//Check Accumulator to see if it is a Logical Operator
+			else if (Accumulator == "and" || Accumulator == "or" || Accumulator == "not") {
+				FilePosition = FilePosition++;
+				return Token(LOGICIAL_OPERATOR, Accumulator);
+			}
+			//Check Accumulator to see if it is a Integer
+			else if (Accumulator == "integer" || Accumulator == "boolean") {
+				FilePosition = FilePosition++;
+				return Token(DATA_TYPE, Accumulator);
+			}
+			//Check Accumulator to see if it is a Boolean
+			else if (Accumulator == "true" || Accumulator == "false") {
+				FilePosition = FilePosition++;
+				return Token(BOOLEAN, Accumulator);
+			}
+			else if (Accumulator == "if" || Accumulator == "than" || Accumulator == "else") {
```

Comment:
neesd to be 'then' not 'than' see here http://www.cs.uni.edu/~wallingf/teaching/cs4550/compiler/specification.html